x-pack/plugins/actions/server/lib/get_execution_log_aggregation.ts

```diff
@@ -0,0 +1,446 @@
+          filter: {
+            bool: {
+              must: [
+                {
+                  match: {
+                    [ACTION_FIELD]: 'execute',
+                  },
+                },
+              ],
+            },
+          },
+          aggs: {
+            executionUuidCardinality: {
+              cardinality: {
+                field: EXECUTION_UUID_FIELD,
+              },
+            },
+          },
+        },
+        executionUuid: {
+          // Bucket by execution UUID
+          terms: {
+            field: EXECUTION_UUID_FIELD,
+            size: DEFAULT_MAX_BUCKETS_LIMIT,
+            order: formatSortForTermSort(sort),
+          },
+          aggs: {
+            // Bucket sort to allow paging through executions
+            executionUuidSorted: {
+              bucket_sort: {
+                sort: formatSortForBucketSort(sort),
+                from: (page - 1) * perPage,
+                size: perPage,
+                gap_policy: 'insert_zeros' as estypes.AggregationsGapPolicy,
+              },
+            },
+            // Filter by action execute doc and get information from this event
+            actionExecution: {
+              filter: {
+                bool: {
+                  must: [
+                    {
+                      match: {
+                        [ACTION_FIELD]: 'execute',
+                      },
+                    },
+                  ],
+                },
+              },
+              aggs: {
+                executeStartTime: {
+                  min: {
+                    field: START_FIELD,
+                  },
+                },
+                scheduleDelay: {
+                  max: {
+                    field: SCHEDULE_DELAY_FIELD,
+                  },
+                },
+                executionDuration: {
+                  max: {
+                    field: DURATION_FIELD,
+                  },
+                },
+                outcomeAndMessage: {
+                  top_hits: {
+                    _source: {
+                      includes: [
+                        OUTCOME_FIELD,
+                        MESSAGE_FIELD,
+                        ERROR_MESSAGE_FIELD,
+                        VERSION_FIELD,
+                        SPACE_ID_FIELD,
+                        ACTION_NAME_FIELD,
+                      ],
+                    },
+                  },
+                },
+              },
+            },
+            // If there was a timeout, this filter will return non-zero doc count
+            timeoutMessage: {
+              filter: {
+                bool: {
+                  must: [
+                    {
+                      match: {
+                        [ACTION_FIELD]: 'execute-timeout',
+                      },
+                    },
+                  ],
+                },
+              },
+            },
+            // Filter out execution UUID buckets where actionExecution doc count is 0
```

Comment:
This might not be necessary since we're filtering out `execute-start` docs at the top level! For the rule exec log, this was filtering out executions that have started (so `execute-start` was written) but haven't finished (no `execute` doc written yet).

Reply:
Should I go ahead and remove this then?

Reply:
Sounds like we don't need it then, at least now.

Reply:
Resolved in this commit